src/hooks/useTransactionList.ts

```diff
@@ -0,0 +1,152 @@
+import { EdgeCurrencyWallet, EdgeTransaction } from 'edge-core-js'
+import * as React from 'react'
+
+import { showError } from '../components/services/AirshipInstance'
+import { useHandler } from './useHandler'
+
+interface Output {
+  atEnd: boolean
+  transactions: EdgeTransaction[]
+  requestMore: () => void
+}
+
+/**
+ * Streams transactions from a wallet.
+ *
+ * Changing the wallet, tokenId, or search term will reset
+ * the transaction list and start the stream from the beginning.
+ * The list will not go all the way to the end at first,
+ * so call the `requestMore` method to request more transactions,
+ * until `atEnd` becomes true.
+ */
+export function useTransactionList(wallet: EdgeCurrencyWallet, tokenId: string | undefined, searchString?: string): Output {
+  const { currencyCode } = tokenId == null ? wallet.currencyInfo : wallet.currencyConfig.allTokens[tokenId]
+
+  const requestMore = React.useRef(() => {})
+
+  // The effect maintains internal mutable state,
+  // and then calls `setOutput` to expose it atomically.
+  const [output, setOutput] = React.useState<Omit<Output, 'requestMore'>>({
+    atEnd: false,
+    transactions: []
+  })
+
+  // This effect runs when the wallet or search term change.
+  // It resets the stream to the beginning each time it runs,
+  // so we don't want it to self-trigger.
+  React.useEffect(() => {
+    // Reset our mutable state to an empty list:
+    let atEnd = false
+    const changedTxs = new Map<string, EdgeTransaction>()
+    const txs: EdgeTransaction[] = []
+
+    // Sends the mutable state to React,
+    // merging the two transaction lists together:
+    function requestRender() {
+      setOutput({
+        atEnd,
+        transactions: txs.map(tx => changedTxs.get(tx.txid) ?? tx)
+      })
+    }
```

Comment:
I like this pattern of using a `requestRender` function because we could always add debouncing in the future if there ever is thrashing of events.